lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -767,6 +770,23 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif querystr:
+        try:
+            vtype = ipaddr(query, 'type')
+            if vtype == 'address':
+                v = ipaddr(query, 'cidr')
+            elif vtype == 'network':
+                v = ipaddr(query, 'subnet')
+
+            query = netaddr.IPNetwork(v)
+        except:
+            return False
+
+        if network_in_network(query, value):
```

Comment:
I believe, you don't even need the if-wrapper, if `query.subnet` won't yield anything, the loop will be "skipped", just like your intention is.